main.go

```diff
@@ -207,6 +221,49 @@ func main() {
 		}
 	}
 
+	if revokeVaultTokensOnUninstall {
```

Comment:
This bit is a little bit hard to parse. We are checking `preDeleteHook` twice, starting a watcher that may not be necessary. I wonder if we could move all of the shutdown handling higher up, maybe to its own function, then have something like:
```
if shutdown:
  shutdown()
else:
   do the normal stuff
```

Reply:
Both the operator and the pre delete hook need the watcher, but they use different contexts. I thought adding a `preDeleteHook` check before starting the watcher would help get away with 2 different watcher setup code blocks.

https://github.com/hashicorp/vault-secrets-operator/blob/5c487a942bbb2b0faaa98faa845b62ef203f1cac/main.go#L230-L233

Reply:
I will look into it but am unsure that we can move much higher up in a neater way since the pre-delete hook needs the client factory.